test/xpu_api/numerics/c.math/cmath.pass.cpp

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// XFAIL: LIBCXX-AIX-FIXME
+
+// <cmath>
+
+#include "support/test_complex.h"
+
+#include <oneapi/dpl/cmath>
+
+// convertible to int/float/double/etc
+template <class T, int N=0>
+struct Value {
+    operator T () { return T(N); }
+};
+
+// See PR21083
+// Ambiguous is a user-defined type that defines its own overloads of cmath
+// functions. When the std overloads are candidates too (by using or adl),
+// they should not interfere.
+struct Ambiguous : std::true_type { // ADL
+    operator float () { return 0.f; }
+    operator double () { return 0.; }
+};
+
+Ambiguous nearbyint(Ambiguous){ return Ambiguous(); }
+
+template <typename HasDoubleSupportInRuntime, typename HasLongDoubleSupportInCompiletime>
+void test_nearbyint()
+{
+    static_assert((std::is_same<decltype(dpl::nearbyint((float)0)), float>::value), "");
+    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((bool)0)), double>::value), ""))
```

Comment:
I suggest using just one macro for lines 38-46. Here and everywhere in such blocks of lines.

Reply:
Fixed